image-loader/app/model/Projector.scala

```diff
@@ -95,11 +95,23 @@ class Projector(config: ImageUploadOpsCfg,
       import ImageIngestOperations.fileKeyFromId
       val s3Key = fileKeyFromId(imageId)
 
-      if (!s3.doesObjectExist(config.originalFileBucket, s3Key))
-        throw new NoSuchImageExistsInS3(config.originalFileBucket, s3Key)
+      val bucket = config.maybeReplicaBucket match {
+        case Some(replicaBucket) if !s3.doesObjectExist(config.originalFileBucket, s3Key) =>
+          replicaBucket
+        case _ =>
+          config.originalFileBucket
+      }
+
+      if (config.maybeReplicaBucket.contains(bucket) && !s3.doesObjectExist(bucket, s3Key)) {
+        logger.error(
+          logMarker,
+          s"Image with id $imageId does not exist at key $s3Key in S3 bucket ${config.originalFileBucket}${config.maybeReplicaBucket.map(rb => s" NOR replica bucket $rb")}"
+        )
+        throw new NoSuchImageExistsInS3(bucket, s3Key)
+      }
```

Comment:
this logic is a bit tough to follow - I think you can be a little adventurous with the match statement and get something nicer:

```suggestion
      val bucket = config.maybeReplicaBucket match {
        case _ if s3.doesObjectExist(config.originalFileBucket, s3Key) =>
          config.originalFileBucket
        case Some(replicaBucket) if s3.doesObjectExist(replicaBucket, s3Key) =>
          replicaBucket
        case _ =>
          logger.error(
            logMarker,
            s"Image with id $imageId does not exist at key $s3Key in S3 bucket ${config.originalFileBucket}${config.maybeReplicaBucket.map(rb => s" NOR replica bucket $rb")}"
          )
          throw new NoSuchImageExistsInS3(config.originalFileBucket, s3Key)
      }
```

not entirely happy with the top case being a `case _` and being dependent on originalFileBucket though - hard to follow in its own way. Possibly it might read a bit better matching on the tuple (config.originalFileBucket, config.maybeReplicaBucket), even if the original is non-Option so a superfluous match?

this block might also work better as a little standalone function?